### DISCUSSION TITLE
How does FreshRSS decide to update a feed?

### DISCUSSION THREAD

**Discussion by rfritz** [FreshRSS/FreshRSS#5663] [Q&A] [answered]
I've got a scraper app aimed at an obscure, complicated web site. The app creates an RSS 2 file, which FreshRSS reads. FreshRSS never updates the feed, yet other feeds work. I suspect I've left something out of my RSS output, but am not sure what. How does FreshRSS decide to update a feed?

**Comment by Frenzie** [comment]
If you run a curl -I you'll see the last-modified header.
```
$ curl -I https://fransdejonge.com/feed/
HTTP/2 200 
vary: Accept-Encoding,Cookie
link: <https://pubsubhubbub.appspot.com>; rel="hub", <https://pubsubhubbub.superfeedr.com>; rel="hub", <https://websubhub.com/hub>; rel="hub", <https://fransdejonge.com/feed/>; rel="self", <https://fransdejonge.com/wp-json/>; rel="https://api.w.org/"
last-modified: Tue, 19 Sep 2023 20:16:01 GMT
etag: "6c10d6f7783f5386d0b1bac9f4e12b07"
cache-control: max-age=5
expires: Wed, 20 Sep 2023 11:14:29 GMT
content-type: application/atom+xml; charset=UTF-8
date: Wed, 20 Sep 2023 11:14:24 GMT
server: Apache
```
Presumably your site says last-modified whenever it was last cached, so no need to update the cache.

**Reply by Alkarex** [reply]
That's a possibility. It could also be that you re-use the same GUID, for instance. You are welcome to share a link or snippet for additional debugging.
You can already check in your Web server logs that the feed is fetched by FreshRSS and with an HTTP 200 response status.
Use also https://validator.w3.org/feed/ to check the validity of your feed.

**Comment by rfritz** [comment, marked as answer]
Thanks, everyone.

My scraper doesn't generate GUIDs at all, and that's probably the problem. Going to be difficult because of the weirdness of the site I'm scraping, but I have some ideas. Hope that FreshRSS doesn't require a GUID to be a permalink.

**Reply by Frenzie** [reply]
A guid can be anything. A (perma)link is used as a fallback if one is missing.

**Comment by rfritz** [comment]
Great, thanks!
-- 
Randolph M. Fritz || ***@***.***

**Comment by rfritz** [comment]
Adding non-permalink GUIDs indeed seems to have fixed the problem. Thank you.

**Comment by Charlotte-br560** [comment]
FreshRSS typically updates feeds based on timestamp changes or ETag/Last-Modified headers in HTTP responses. If your feed isn't updating, double-check your RSS output for any missing or incorrect metadata.
